### ISSUE TITLE
Improve the performances and the memory

### ISSUE THREAD

**Issue by mgrr** [nikic/PHP-Parser#1044]
Why you dont use the \Generator in php to replace the array stmts ?

**Comment by theofidry** [comment]
You mean typing `Generator` in favor of `array` as a return type? Would that really help with performances and memory management?

**Comment by nikic** [comment]
Yeah, I don't think generators are particularly useful for a parser. I think the only area where they might be useful is tokenization, and there we get the full array from PHP anyway. I don't think there's a use case for making the parser itself lazy.
